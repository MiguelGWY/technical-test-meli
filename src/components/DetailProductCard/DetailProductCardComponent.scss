.container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;

  &__product-detail {
    display: flex;
  }

  &__img-description {
    flex-basis: 0%;
    flex-grow: 1;
    flex-shrink: 1;
    padding: 10px;
  }

  &__img img {
    object-fit: contain;
    height: 500px;
    width: 500px;
  }

  &__title-description {
    font-size: 24px;
    font-weight: 400;
    padding-bottom: 24px;
  }

  &__description {
    color: #666;
    font-size: 20px;
  }

  &__product-info {
    border: thin solid #dadada;
    border-radius: 5px;
    padding: 10px;
  }

  &__condition {
    color: rgba(0, 0, 0, 0.55);
    font-size: 14px;
    font-weight: 400;
    white-space: pre-wrap;
  }

  &__product-title {
    font-size: 22px;
    margin-bottom: 8px;
    margin-right: 28px;
    padding: 0;
  }

  &__price {
    font-size: 36px;
    font-weight: 300;
    margin-top: 15px;
    margin-bottom: 0;
  }

  &__subtitles {
    line-height: 1;
    color: #666;
    margin-top: 3px;
  }

  &__buy-button {
    background-color: #2968c8;
    color: white;
    padding: 10px 20px;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    width: 100%;
    margin-top: 30px;

    &:hover {
      background-color: #2f82ff;
    }
  }
}
