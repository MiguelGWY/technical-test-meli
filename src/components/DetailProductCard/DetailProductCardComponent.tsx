import { formatCondition } from "../../utils/formattedCondition";
import "./DetailProductCardComponent.scss";

export const DetailProductCard = (item: Item) => {
  const nameSpace = "container";

  return (
    <div className={nameSpace}>
      <div className={`${nameSpace}__product-detail`}>
        <div className={`${nameSpace}__img-description`}>
          <div className={`${nameSpace}__img`}>
            <img src={item.picture} alt="product-image" />
          </div>
          <h2 className={`${nameSpace}__title-description`}>
            Descripcion del producto
          </h2>
          <p className={`${nameSpace}__description`}>{item.description}</p>
        </div>
        <div className={`${nameSpace}__product-info`}>
          <div className={`${nameSpace}__condition`}>
            {`${formatCondition(item.condition)} | +100 vendidos`}
          </div>
          <div className={`${nameSpace}__product-title`}>{item.title}</div>
          <p className={`${nameSpace}__price`}>$ {item.price.amount}</p>
          <p className={`${nameSpace}__subtitles`}>en 48x $ 15.685</p>
          <button className={`${nameSpace}__buy-button`}>Comprar</button>
        </div>
      </div>
    </div>
  );
};
