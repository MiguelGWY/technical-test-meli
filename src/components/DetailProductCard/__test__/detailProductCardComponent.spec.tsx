import { render } from "@testing-library/react";
import { DetailProductCard } from "../DetailProductCardComponent";
import "@testing-library/jest-dom";

const itemMock = {
  id: "1",
  title: "test product",
  price: {
    amount: 1234,
    decimals: 0,
    currency: "ARG",
  },
  thumbnail: "www.ddd.com",
  picture: "www.ddd.com",
  condition: "new",
  free_shipping: false,
  sold_quantity: 304,
  description: "Description Item",
  categories: [""],
  category_id: "112DADSAD",
  currency_id: "SDDDWEEEF",
};

describe("DetailProductCardComponent", () => {
  it("should render DetailProductCardComponent", () => {
    const { container } = render(<DetailProductCard {...itemMock} />);
    expect(container).toMatchSnapshot();
  });

  it("should display the product condition in Spanish when it is new", () => {
    const {getByText} = render(<DetailProductCard {...itemMock} />);
    expect(getByText("Nuevo | +100 vendidos")).toBeInTheDocument();
  });
});
