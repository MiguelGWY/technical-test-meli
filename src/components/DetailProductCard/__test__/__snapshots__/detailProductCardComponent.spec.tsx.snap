// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailProductCardComponent should render DetailProductCardComponent 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container__product-detail"
    >
      <div
        class="container__img-description"
      >
        <div
          class="container__img"
        >
          <img
            alt="product-image"
            src="www.ddd.com"
          />
        </div>
        <h2
          class="container__title-description"
        >
          Descripcion del producto
        </h2>
        <p
          class="container__description"
        >
          Description Item
        </p>
      </div>
      <div
        class="container__product-info"
      >
        <div
          class="container__condition"
        >
          Nuevo | +100 vendidos
        </div>
        <div
          class="container__product-title"
        >
          test product
        </div>
        <p
          class="container__price"
        >
          $ 
          1234
        </p>
        <p
          class="container__subtitles"
        >
          en 48x $ 15.685
        </p>
        <button
          class="container__buy-button"
        >
          Comprar
        </button>
      </div>
    </div>
  </div>
</div>
`;
